test/Splitter.t.sol

```diff
@@ -0,0 +1,433 @@
+        checkAuth(address(splitter), "Splitter");
+    }
+
+    function testAuthModifiers() public virtual {
+        assert(splitter.wards(address(this)) == 0);
+
+        checkModifier(address(splitter), string(abi.encodePacked("Splitter", "/not-authorized")), [
+            Splitter.kick.selector,
+            Splitter.cage.selector
+        ]);
+    }
+
+    function testFileUint() public {
+        checkFileUint(address(splitter), "Splitter", ["burn"]);
+    }
+
+    function testFileAddress() public {
+        checkFileAddress(address(splitter), "Splitter", ["flapper"]);
+    }
+
+    function testVatCanAfterFile() public {
+        assertEq(vat.can(address(splitter), address(0xf1)), 0);
+
+        vm.prank(PAUSE_PROXY); splitter.file("flapper", address(0xf1));
+
+        assertEq(vat.can(address(splitter), address(0xf1)), 1);
+        assertEq(vat.can(address(splitter), address(0xf2)), 0);
+
+        vm.prank(PAUSE_PROXY); splitter.file("flapper", address(0xf2));
+
+        assertEq(vat.can(address(splitter), address(0xf1)), 0);
+        assertEq(vat.can(address(splitter), address(0xf2)), 1);
+    }
+
+    function testKick() public {
+        doKick();
+    }
+
+    function testKickBurnOnly() public {
+        doKick();
+        vm.warp(block.timestamp + flapper.hop());
+
+        vm.prank(PAUSE_PROXY); splitter.file("burn", WAD);
+
+        // make sure the slippage of the first kick doesn't block us
+        uint256 _marginalWant = marginalWant(MKR, address(medianizer));
+        vm.prank(PAUSE_PROXY); flapper.file("want", _marginalWant * 99 / 100);
+        doKick();
+    }
+
+    function testKickAfterHop() public {
+        doKick();
+        vm.warp(block.timestamp + flapper.hop());
+
+        // make sure the slippage of the first kick doesn't block us
+        uint256 _marginalWant = marginalWant(MKR, address(medianizer));
+        vm.prank(PAUSE_PROXY); flapper.file("want", _marginalWant * 99 / 100);
+        doKick();
+    }
+
+    function testKickBeforeHop() public {
+        doKick();
+        vm.warp(block.timestamp + flapper.hop() - 1 seconds);
+
+        // make sure the slippage of the first kick doesn't block us
+        uint256 _marginalWant = marginalWant(MKR, address(medianizer));
+        vm.prank(PAUSE_PROXY); flapper.file("want", _marginalWant * 99 / 100);
+        vm.expectRevert("FlapperUniV2SwapOnly/kicked-too-soon");
+        vow.flap();
+    }
+
+    function testKickFlapperNotSet() public {
+        vm.prank(PAUSE_PROXY); splitter.file("flapper", address(0));
+        vm.expectRevert(bytes(""));
+        vow.flap();
+    }
+
+    function checkChangeRewardDuration(uint256 newDuration) private {
+        uint256 topup = 5707 * (WAD - 70 * WAD / 100);
+        doKick();
+        assertEq(farm.rewardsDuration(), flapper.hop());
+        assertEq(farm.rewardsDuration(), 30 minutes);
+        assertEq(farm.rewardRate(), topup / 30 minutes);
+        uint256 prevRewardRate = farm.rewardRate();
```

Comment:
```suggestion
        uint256 prevRewardRate = farm.rewardRate();
        assertEq(prevRewardRate, topup / 30 minutes);
```

Reply:
Slight preference to keeping it as-is as semantically the reward rate is not yet the "previous" reward rate when it is being checked here.